flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -64,6 +66,15 @@ public MetricRegistry(Configuration config) {
 		}
 		this.scopeFormats = scopeFormats;
 
+		char delim;
+		try {
+			delim = config.getString(ConfigConstants.METRICS_SCOPE_DELIMITER, ".").charAt(0);
```

Comment:
well...sure it would work, but it will only be a matter of time until someone forgets to pass a delimiter and gets the weirdest metric names imaginable. I would simply *prefer* a solution that can catch this error at compile-time.